src/promptflow/promptflow/_sdk/operations/_trace_operations.py

```diff
@@ -13,18 +16,42 @@ class TraceOperations:
     def list_spans(
         self,
         session_id: typing.Optional[str] = None,
-        parent_span_id: typing.Optional[str] = None,
     ) -> typing.List[Span]:
         orm_spans = ORMSpan.list(
             session_id=session_id,
-            parent_span_id=parent_span_id,
         )
         return [Span._from_orm_object(orm_span) for orm_span in orm_spans]
 
     def list_line_runs(
         self,
         session_id: typing.Optional[str] = None,
     ) -> typing.List[LineRun]:
+        line_runs = []
         orm_spans_group_by_trace_id = ORMLineRun.list(session_id=session_id)
-        spans = [Span._from_orm_object(orm_span) for orm_spans in orm_spans_group_by_trace_id for orm_span in orm_spans]
-        return LineRun._from_spans(spans)
+        # merge spans with same `line_run_id` or `referenced.line_run_id` (if exists)
+        grouped_orm_spans = {}
+        for orm_spans in orm_spans_group_by_trace_id:
+            first_orm_span = orm_spans[0]
+            attributes = json.loads(first_orm_span.content)[SpanFieldName.ATTRIBUTES]
+            if (
+                SpanAttributeFieldName.LINE_RUN_ID not in attributes
+                and SpanAttributeFieldName.REFERENCED_LINE_RUN_ID not in attributes
+            ):
+                # no `line_run_id` or `referenced.line_run_id` in attributes
+                # standard OpenTelemetry trace, regard as a line run
+                grouped_orm_spans[first_orm_span.trace_id] = copy.deepcopy(orm_spans)
+            elif SpanAttributeFieldName.LINE_RUN_ID in attributes:
+                # for other traces, group by `line_run_id` as a line run
+                line_run_id = attributes[SpanAttributeFieldName.LINE_RUN_ID]
+                if line_run_id not in grouped_orm_spans:
+                    grouped_orm_spans[line_run_id] = []
+                grouped_orm_spans[line_run_id].extend(copy.deepcopy(orm_spans))
+            else:
+                referenced_line_run_id = attributes[SpanAttributeFieldName.REFERENCED_LINE_RUN_ID]
```

Comment:
We may skip this branch.

Reply:
good catch, updated the condition and skip some cases.